news/2020-09-15-CDI_Lite_extension.adoc

```diff
@@ -0,0 +1,284 @@
+---
+title: "Build Compatible Extensions Proposal"
+summary: Description of an extension SPI for CDI lite
+layout: post
+authors: [ladt]
+tags: [Proposal]
+desc: Build Compatible Extensions Proposal
+priority: 1
+change_frequency: daily
+---
+
+
+A few months back we shared our vision for CDI lite.
+To make it short the goal with CDI lite is to make the spec cloud native. This will be done by identifying a features' subset that can be implemented with a build time only approach (no more runtime proxies and other “magic” requirements) .
+
+The idea is to keep the CDI programming model everyone loves but allow implementation to produce lighter and faster binaries.
+
+One issue we raised in the last post is that this build-time approach is not compatible with CDI portable extension. So, with current CDI features we can only imagine CDI Lite without extension mechanism. Knowing the great popularity of portable extensions, it was a capability we needed to revisit in our initial proposal.
+
+It’s the reason why we worked to propose a new extension mechanism for CDI Lite. Today we are sharing this proposal with you as it is still a work in progress but seems ready to be a strong starting point to design an extension mechanism for CDI lite.
+
+When we started thinking about a lite API design space, we mostly constrained ourselves to one particularly common use case: annotation transformations. Many things that Portable Extensions allow can be achieved simply by adding an annotation here or removing an annotation there. We know that other transformations are necessary, but the API for annotation transformation is the most developed proposal we have. The other ones are significantly more rough, so bear with us please and submit feedback! (*TODO* where?)
```

Comment:
TODO for feedback...

Reply:
The issue isn't created yet (I suppose we'll be creating that shortly after posting this).